include/odp/api/spec/traffic_mngr.h

```diff
@@ -1655,6 +1655,20 @@ int odp_tm_queue_disconnect(odp_tm_queue_t tm_queue);
  */
 int odp_tm_enq(odp_tm_queue_t tm_queue, odp_packet_t pkt);
 
+/** The odp_tm_enq_multi() function is used to add packets to a given TM system.
+ * This API takes multiple packets to enqueue unlike odp_tm_enq() but
+ * rest of the behaviour is identical to that of odp_tm_enq().
+ *
+ * @param tm_queue  Specifies the tm_queue (and indirectly the TM system).
+ * @param packets   Array of packets to enqueue.
+ * @param num       Number of packets to send.
+ * @return          Returns >0 upon success indicating number of packets
+ *                  enqueued, <= 0 upon failure. One of the more common
+ *                  failure reasons is WRED drop.
```

Comment:
I wonder how this is supposed to be used with WRED. How should the function behave when WRED wants to drop the 5th packet out of a 10 packet batch? Comsume all 10 packets, dropping the 5th one, and return 10? Consume all 10 packets, dropping the 5th one, and return 9? Consume 4 packets and return 4 and then return failure the next time the app calls the function (but maybe we probabilistic drop decision is not the same anymore)?

So this clearly needs more detailed specification on which packets get consumed, how (if?) WRED drop gets indicated etc.

Reply:
My guess is for WRED, stats should be in different mechanism and we should report all the 10 consumed if space is there even though some packets get dropped due to WRED.
However in our HW, we don't support WRED so, I cannot really comment here concretely

Reply:
I think it would make sense to just consume all packets if (W)RED is used and then have (W)RED drop the ones it wants. I am not sure if tail drop should work the same way. Maybe.